### PR TITLE
Autorun migrations in community edition

### LINKED ISSUE #4680: Run migrations in community edition automatically
Will land this in v0.68.0, the expectation will be that any backups are done before updating the software. Required migrations will run automatically.

### PR DESCRIPTION
**Pull request by apoorv-mishra**
Closes #4680

### CODE REVIEW

File: server/utils/startup.ts [hunk 1]
```diff
@@ -20,32 +21,74 @@ export function checkPendingMigrations() {
...
+function logMigrations(migrations: string[]) {
+  Logger.warn("You have pending migrations");
+  Logger.warn(
+    migrations
+      .map((line, i) => `${i + 1}. ${line.replace("down ", "")}`)
+      .join("\n")
+  );
+  Logger.warn(
+    "Please run `yarn db:migrate` or `yarn db:migrate --env production-ssl-disabled` to run all pending migrations"
+  );
```
Comment (tommoor) [inline server/utils/startup.ts:75]:
```suggestion
```

It's ok, I don't need this and this is only a safety measure for cloud now.

File: server/utils/startup.ts [hunk 2]
```diff
@@ -20,32 +21,74 @@ export function checkPendingMigrations() {
...
+    const cmdResult = execSync(
+      `yarn db:migrate${
+        env.PGSSLMODE === "disable" ? " --env=production-ssl-disabled" : ""
+      }`
+    );
+    const cmdOutput = Buffer.from(cmdResult).toString("utf-8");
+    Logger.info("database", cmdOutput);
+    Logger.info("database", "Done.");
   } catch (err) {
+    Logger.warn(chalk.red("Failed to run migrations!"));
```
Comment (tommoor) [inline server/utils/startup.ts:52]:
```suggestion
```

This is resulting in a double error, seems fine without?
image

**Review by tommoor — APPROVED** [review]

**Comment by tommoor** [comment]
Oh, I meant to add we should remove the `db:migrate` call from within the `Makefile` as it'll be run twice now.

File: server/utils/startup.ts (hunk 2 above)
Comment (apoorv-mishra) [reply server/utils/startup.ts:52]: I believe removing `db:migrate` from `Makefile` resolves this. No further action needed?

Comment (tommoor) [reply server/utils/startup.ts:52]: I don't think it changes it to be honest, I'm easy either way here

Comment (apoorv-mishra) [reply server/utils/startup.ts]: The first error is logged from within the `sequelize` cli :see_no_evil: 

Catching errors centrally within `checkMigrations` now.

**Review by tommoor — APPROVED** [review]

**Comment by patzm** [comment]
awesome feature!!! Thanks @apoorv-mishra, this has always been a pain for me. In particular in a `docker-compose` stack, in which the _just run a command_ thinking isn't really native.

**Pull request merged**